fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/system/ChunkReader.java

```diff
@@ -146,70 +166,59 @@ private void fillChunkDataBuffer(List<Resource> resources) throws Exception {
+                    indexOfCurrentTypeFilter = 0;
+                    isDoDuplicationCheck = false;
+                    loadedResourceIds.clear();
+                }
             } else {
-                // More resource types to read, so reset pageNum, partNum and move resource type index to the next.
+             // If there is more typeFilter to process for current resource type, then reset pageNum only and move to the next typeFilter.
                 pageNum = 1;
-                chunkData.setPartNum(1);
-                indexOfCurrentResourceType++;
+                indexOfCurrentTypeFilter++;
             }
         }
-        if (fhirTenant == null) {
-            fhirTenant = Constants.DEFAULT_FHIR_TENANT;
-            logger.fine("readItem: Set tenant to default!");
-        }
-        if (fhirDatastoreId == null) {
-            fhirDatastoreId = Constants.DEFAULT_FHIR_TENANT;
-            logger.fine("readItem: Set DatastoreId to default!");
-        }
-        if (fhirSearchPageSize != null) {
-            try {
-                pageSize = Integer.parseInt(fhirSearchPageSize);
-                logger.fine("readItem: Set page size to " + pageSize + ".");
-            } catch (Exception e) {
-                logger.warning("readItem: Set page size to default(" + Constants.DEFAULT_SEARCH_PAGE_SIZE + ").");
-            }
-        }
-
-        if (cosBucketObjectName != null
-                && cosBucketObjectName.trim().length() > 0
-                // Single COS object uploading is for single resource type export only.
-                && ResourceTypes.size() == 1) {
-            isSingleCosObject = true;
-            logger.info("readItem: Use single COS object for uploading!");
-        }
 
-        FHIRRequestContext.set(new FHIRRequestContext(fhirTenant, fhirDatastoreId));
-        FHIRPersistenceHelper fhirPersistenceHelper = new FHIRPersistenceHelper();
-        FHIRPersistence fhirPersistence = fhirPersistenceHelper.getFHIRPersistenceImplementation();
         Class<? extends Resource> resourceType = ModelSupport
-                .getResourceType(ResourceTypes.get(indexOfCurrentResourceType));
+                .getResourceType(resourceTypes.get(indexOfCurrentResourceType));
         FHIRSearchContext searchContext;
         FHIRPersistenceContext persistenceContext;
         Map<String, List<String>> queryParameters = new HashMap<>();
 
-        List<String> searchCreteria = new ArrayList<String>();
+        // Add the search parameters from the current typeFilter for current resource type.
+        if (searchParametersForResoureTypes.get(resourceType) != null) {
+            queryParameters.putAll(searchParametersForResoureTypes.get(resourceType).get(indexOfCurrentTypeFilter));
+            if (searchParametersForResoureTypes.get(resourceType).size() > 1) {
+                isDoDuplicationCheck = true;
+            }
+        }
+
+        List<String> searchCriteria = new ArrayList<String>();
```

Comment:
just a thought, maybe put this in a utility class to process the query parameters

Reply:
en... good thought! will consider in the next big change for export jobs - issue #779 partitionize the export jobs.

Reply:
yeah - no worries - this was just a thought